### PR TITLE
qa_crowbarsetup: add test for WSGI services reload

### PR DESCRIPTION
**Pull request by aplanas**

### CODE REVIEW

**Comment by nicolasbock** [comment]
@aplanas Is the idea to see whether those services are really running behind apache?

**Comment by aplanas** [comment]
@nicolasbock the idea is to be sure that a reload of apache is, indeed, a reload of those services. This is mostly to be sure that changes like https://github.com/crowbar/crowbar-openstack/pull/830 do not break the cloud

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -4007,10 +4007,17 @@ function oncontroller_testsetup
...
+    fi
+
     echo "RadosGW Tests: $radosgwret"
     echo "Tempest: $tempestret"
     echo "Volume in VM: $volumeresult"
     echo "Ports in binding_failed: $portresult"
+    echo "WSGI service reload: $serviceresult"
 
     test $tempestret = 0 -a $volumecreateret = 0 -a $volumeattachret = 0 \
         -a $radosgwret = 0 -a $portresult = 0 || exit 102
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: should it be red on failure? then add a check here.

Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: indeed!

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -4306,6 +4313,79 @@ function oncontroller_testpostupgrade
...
+        *)
+            ;;
+        esac
+
+        local ret=0
+
+        # Is the service up and running?
+        echo -n "Testing $service WSGI service initial state: "
+        $command &>/dev/null
+        if [ $? -ne 0 ]; then
```
Comment (bmwiedemann) [inline scripts/qa_crowbarsetup.sh]: you could do `if $command` ; then
(and put the OK part first)

**Comment by bmwiedemann** [comment]
shouldn't that have detected
https://bugzilla.suse.com/show_bug.cgi?GoAheadAndLogIn=1&id=1024713 ?
or do we have fixes in develcloud7 that are not in GMC?

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: Done!

**Comment by aplanas** [comment]
@bmwiedemann Not sure, I didn't deployed keystone with SSL, but I am sure that the reload of keystone is detected in line 4360, that is where I broke keystone.

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -4306,6 +4313,76 @@ function oncontroller_testpostupgrade
...
+            echo "ERROR"
+            ret=302
+        fi
+
+        # Recover the service
+        mv $cfg.bak $cfg
+        systemctl reload apache2.service
+
+        # Is the service up and running?
+        echo -n "Testing $service WSGI service reload in success state: "
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: code duplication, might be worth a function

File: scripts/qa_crowbarsetup.sh [hunk 4]
```diff
@@ -4306,6 +4313,76 @@ function oncontroller_testpostupgrade
...
+            command="ceilometer sample-list"
+            ;;
+        keystone)
+            command="openstack user list"
+            ;;
+        horizon)
+            cfg="/srv/www/openstack-dashboard/openstack_dashboard/local/local_settings.d/_100_local_settings.py"
+            command="curl --location --fail http://localhost"
+            ;;
+        *)
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: two useless lines

File: scripts/qa_crowbarsetup.sh [hunk 5]
```diff
@@ -4007,13 +4007,20 @@ function oncontroller_testsetup
         done
     fi
 
+    if iscloudver 7plus ; then
+        # check that the services behing WSGI can be reloaded
+        oncontroller_testservices
+        serviceresult=$?
```
Comment (jdsn) [inline scripts/qa_crowbarsetup.sh]: missing: `local serviceresult`

Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: done

File: scripts/qa_crowbarsetup.sh [hunk 6]
```diff
@@ -4306,6 +4313,76 @@ function oncontroller_testpostupgrade
     echo "test post-upgrade successful."
 }
 
+function oncontroller_testservices
+{
+    local services=(aodh ceilometer barbican keystone horizon)
+    for service in "${services[@]}"; do
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: done

File: scripts/qa_crowbarsetup.sh [hunk 7]
```diff
@@ -4306,6 +4313,76 @@ function oncontroller_testpostupgrade
...
+        local cfg=/etc/$service/$service.conf.d/100-$service.conf
+
+        # Discover the command line to execute and how to make it fail
+        local command="false"
```
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: ah no, look above

File: scripts/qa_crowbarsetup.sh (hunk 4 above)
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: uhmm done

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (aplanas) [inline scripts/qa_crowbarsetup.sh]: the return is different, and for 2 lines of code I prefer to be explicit here.

**Comment by bmwiedemann** [comment]
maybe rebase to get an updated testrun

**Comment by aplanas** [comment]
rebased

**Pull request closed without merging**